### PR TITLE
Feature/add note screen

### PR DESCRIPTION
**Pull request by katerynaYaryfa**

### CODE REVIEW

File: lib/bottom_navigation_bar.dart [hunk 1]
```diff
@@ -1,92 +1,153 @@
 import 'package:flutter/material.dart';
 import 'package:flutter_svg/svg.dart';
+import 'package:mood_tracker/new_note_screen/add_new_note_screen.dart';
+import 'package:mood_tracker/settings_screen/settings_screen.dart';
 
-class CustomNavigationBar extends StatelessWidget {
+class CustomNavigationBar extends StatefulWidget {
```
Comment (VovaMedvediev) [inline lib/bottom_navigation_bar.dart:6]: Refactor the whole widget in order to use BottomNavigationBar widget with custom FAB implementation (UI won't change, only code)

File: lib/new_note_screen/add_new_note_screen.dart [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+import 'package:flutter_svg/svg.dart';
+import 'package:mood_tracker/new_note_screen/how_was_your_day_widget.dart';
+import 'package:mood_tracker/new_note_screen/photo_of_the_day_widget.dart';
+import 'package:mood_tracker/new_note_screen/select_categories_screen.dart';
+import 'package:mood_tracker/provider/settings_notifier.dart';
+import 'package:provider/provider.dart';
+
+import 'day_in_one_sentence_widget.dart';
+
+class AddNewNoteScreen extends StatefulWidget {
```
Comment (VovaMedvediev) [inline lib/new_note_screen/add_new_note_screen.dart:11]: There is no need to use StatefulWidget, refactor to StatelessWidget

File: lib/new_note_screen/add_new_note_screen.dart [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+        bottomOpacity: 0.0,
+        elevation: 0.0,
+      ),
+      body: SingleChildScrollView(
+        child: Container(
+          width: double.infinity,
+          child: Column(
+            children: [
+              const HowWasYourDayWidget(),
+              const SizedBox(
```
Comment (VovaMedvediev) [inline lib/new_note_screen/add_new_note_screen.dart:96]: Add spacers.dart file with custom spacers, e.g:
class SpaceH16 extends StatelessWidget {
build() {
return SizedBox(height: 16);
}
}

And use such widgets in all places

File: lib/new_note_screen/add_new_note_screen.dart [hunk 4]
```diff
@@ -0,0 +1,111 @@
...
+                height: 16,
+              ),
+              const DayInOneSentenceWidget(),
+              const SizedBox(
+                height: 16,
+              ),
+              PhotoOfTheDayWidget(
+                  images: images, pickImage: pickImage, imageSize: imageSize)
```
Comment (VovaMedvediev) [inline lib/new_note_screen/add_new_note_screen.dart:104]: add trailing comma

File: lib/new_note_screen/add_new_note_screen.dart [hunk 5]
```diff
@@ -0,0 +1,111 @@
...
+
+class _AddNewNoteScreenState extends State<AddNewNoteScreen> {
+  @override
+  Widget build(BuildContext context) {
+    final screenWidth = MediaQuery.of(context).size.width;
+    const imagePadding = 64;
+    final imageSize = screenWidth - imagePadding;
+    final images = context.watch<SettingsNotifier>().images;
+    final pickImage = context.watch<SettingsNotifier>().pickImage;
+    print('_______AddNewNoteScreen____images $images');
```
Comment (VovaMedvediev) [inline lib/new_note_screen/add_new_note_screen.dart:26]: delete print please

File: lib/new_note_screen/day_in_one_sentence_widget.dart [hunk 6]
```diff
@@ -0,0 +1,86 @@
...
+  const DayInOneSentenceWidget({
+    Key? key,
+  }) : super(key: key);
+
+  @override
+  Widget build(BuildContext context) {
+    return Container(
+      padding: const EdgeInsets.all(16),
+      margin: EdgeInsets.symmetric(horizontal: 16.0),
+      // height: 110,
```
Comment (VovaMedvediev) [inline lib/new_note_screen/day_in_one_sentence_widget.dart:14]: delete unused code please

File: lib/new_note_screen/day_in_one_sentence_widget.dart [hunk 7]
```diff
@@ -0,0 +1,86 @@
...
+        crossAxisAlignment: CrossAxisAlignment.start,
+        children: [
+          const Text(
+            'Day in one sentence',
+            style: TextStyle(
+              fontSize: 16,
+              fontWeight: FontWeight.bold,
+            ),
+          ),
+          SizedBox(
```
Comment (VovaMedvediev) [inline lib/new_note_screen/day_in_one_sentence_widget.dart:39]: replace with SpaceH16()

File: lib/new_note_screen/day_in_one_sentence_widget.dart [hunk 8]
```diff
@@ -0,0 +1,86 @@
...
+            height: 16,
+          ),
+          TextField(
+            keyboardType: TextInputType.multiline,
+            // maxLines: null,
```
Comment (VovaMedvediev) [inline lib/new_note_screen/day_in_one_sentence_widget.dart:44]: delete unused code

File: lib/new_note_screen/day_in_one_sentence_widget.dart [hunk 9]
```diff
@@ -0,0 +1,86 @@
...
+            inputFormatters: [
+              LengthLimitingTextInputFormatter(2000),
```
Comment (VovaMedvediev) [inline lib/new_note_screen/day_in_one_sentence_widget.dart:46]: add private const variable for 2000 value with understandable name. E.g _maxCharacterLength or something similar

File: lib/new_note_screen/day_in_one_sentence_widget.dart [hunk 10]
```diff
@@ -0,0 +1,86 @@
...
+              filled: true,
+              isCollapsed: true,
+              fillColor: Color(0xFFF6FAFB),
+              labelText: 'Enter your thoughts',
+              floatingLabelBehavior: FloatingLabelBehavior.never,
+              labelStyle: TextStyle(
+                color: Color(0xFFD1D4DE),
+              ),
+              border:
+                  // InputBorder.none
```
Comment (VovaMedvediev) [inline lib/new_note_screen/day_in_one_sentence_widget.dart:71]: delete unused code

File: lib/new_note_screen/how_was_your_day_widget.dart [hunk 11]
```diff
@@ -0,0 +1,202 @@
...
+  const HowWasYourDayWidget({
+    Key? key,
+  }) : super(key: key);
+
+  @override
+  State<HowWasYourDayWidget> createState() => _HowWasYourDayWidgetState();
+}
+
+class _HowWasYourDayWidgetState extends State<HowWasYourDayWidget> {
+  bool value1 = true;
```
Comment (VovaMedvediev) [inline lib/new_note_screen/how_was_your_day_widget.dart:13]: Refactor to approach with Enum (e.g BottomNavigation implementation, but replace int index with enum index).
Move this logic to ChangeNotifier

File: lib/new_note_screen/how_was_your_day_widget.dart [hunk 12]
```diff
@@ -0,0 +1,202 @@
...
+  bool value6 = true;
+
+  @override
+  Widget build(BuildContext context) {
+    print('$value4');
+
+    return Container(
+      padding: const EdgeInsets.all(16),
+      margin: EdgeInsets.symmetric(horizontal: 16.0),
+      // height: 110,
```
Comment (VovaMedvediev) [inline lib/new_note_screen/how_was_your_day_widget.dart:27]: delete unused code

File: lib/new_note_screen/how_was_your_day_widget.dart [hunk 13]
```diff
@@ -0,0 +1,202 @@
...
+            ),
+          ),
+          const SizedBox(
+            height: 16,
+          ),
+          Container(
+            child: Row(
+              mainAxisAlignment: MainAxisAlignment.center,
+              children: [
+                InkWell(
```
Comment (VovaMedvediev) [inline lib/new_note_screen/how_was_your_day_widget.dart:57]: Refactor to general widget (MoodIcon or something similar)

File: lib/new_note_screen/photo_of_the_day_widget.dart [hunk 14]
```diff
@@ -0,0 +1,249 @@
...
+  }) : super(key: key);
+
+  final List<File> images;
+  final Future Function() pickImage;
+  final double imageSize;
+
+  @override
+  Widget build(BuildContext context) {
+    return Container(
+      padding: EdgeInsets.only(top: 16, right: 16, left: 16, bottom: 16),
```
Comment (VovaMedvediev) [inline lib/new_note_screen/photo_of_the_day_widget.dart:24]: comma

File: lib/new_note_screen/photo_of_the_day_widget.dart [hunk 15]
```diff
@@ -0,0 +1,249 @@
...
+      // height: 364,
```
Comment (VovaMedvediev) [inline lib/new_note_screen/photo_of_the_day_widget.dart:25]: remove unused code

File: lib/new_note_screen/photo_of_the_day_widget.dart [hunk 16]
```diff
@@ -0,0 +1,249 @@
...
+      margin: const EdgeInsets.symmetric(horizontal: 16.0),
```
Comment (VovaMedvediev) [inline lib/new_note_screen/photo_of_the_day_widget.dart:26]: comma

File: lib/new_note_screen/photo_of_the_day_widget.dart [hunk 17]
```diff
@@ -0,0 +1,249 @@
...
+              Text(
+                'Photo of the day',
+                style: TextStyle(
+                  fontSize: 16,
+                  fontWeight: FontWeight.bold,
+                ),
+              ),
+              if (images.length == 3)
+                SvgPicture.asset('images/addNewImage.svg',
+                    color: const Color(0xFFFFBAB0)),
```
Comment (VovaMedvediev) [inline lib/new_note_screen/photo_of_the_day_widget.dart:55]: comma

File: lib/new_note_screen/photo_of_the_day_widget.dart [hunk 18]
```diff
@@ -0,0 +1,249 @@
...
+          if (images.length == 1)
+            AddNewImage(
+              left: 19,
+              top: 258,
+              image: images[0],
+              height: 296,
+              width: 296,
+              cacheHeight: 296,
+              cacheWidth: 296,
+              // vertical: 258,
```
Comment (VovaMedvediev) [inline lib/new_note_screen/photo_of_the_day_widget.dart:80]: unused code

File: lib/new_note_screen/photo_of_the_day_widget.dart [hunk 19]
```diff
@@ -0,0 +1,249 @@
...
+              children: [
+                AddNewImage(
+                  left: 11,
+                  top: 110,
+                  height: 140,
+                  width: 140,
+                  image: images[0],
+                  cacheHeight: 140,
+                  cacheWidth: 140,
+                  // vertical: 110,
```
Comment (VovaMedvediev) [inline lib/new_note_screen/photo_of_the_day_widget.dart:94]: unused code

File: lib/new_note_screen/photo_of_the_day_widget.dart [hunk 20]
```diff
@@ -0,0 +1,249 @@
...
+                ),
+              ),
+            )
+        ],
+      ),
+    );
+  }
+}
+
+class AddNewImage extends StatelessWidget {
```
Comment (VovaMedvediev) [inline lib/new_note_screen/photo_of_the_day_widget.dart:189]: extract to separate widget

File: lib/new_note_screen/photo_of_the_day_widget.dart [hunk 21]
```diff
@@ -0,0 +1,249 @@
...
+  const AddNewImage({
+    required this.height,
+    required this.image,
+    required this.width,
+    required this.cacheHeight,
+    required this.cacheWidth,
+    required this.left,
+    required this.top,
+    // required this.bottom,
```
Comment (VovaMedvediev) [inline lib/new_note_screen/photo_of_the_day_widget.dart:198]: unused code

File: lib/new_note_screen/photo_of_the_day_widget.dart [hunk 22]
```diff
@@ -0,0 +1,249 @@
...
+    Key? key,
+  }) : super(key: key);
+  final double width;
+  final double height;
+  final File image;
+  final int cacheHeight;
+  final int cacheWidth;
+  final double left;
+  final double top;
+  // final double bottom;
```
Comment (VovaMedvediev) [inline lib/new_note_screen/photo_of_the_day_widget.dart:208]: unused code

File: lib/new_note_screen/photo_of_the_day_widget.dart [hunk 23]
```diff
@@ -0,0 +1,249 @@
+import 'dart:io';
+
+import 'package:flutter/material.dart';
+import 'package:flutter_svg/svg.dart';
+import 'package:provider/provider.dart';
+
+import '../provider/settings_notifier.dart';
+
+class PhotoOfTheDayWidget extends StatelessWidget {
```
Comment (VovaMedvediev) [inline lib/new_note_screen/photo_of_the_day_widget.dart:9]: discuss it with me in dm please

File: lib/new_note_screen/select_categories_screen.dart [hunk 24]
```diff
@@ -0,0 +1,116 @@
+import 'dart:io';
+
+import 'package:flutter/material.dart';
+import 'package:flutter/services.dart';
+import 'package:image_picker/image_picker.dart';
+
+class SelectCategoriesScreen extends StatefulWidget {
```
Comment (VovaMedvediev) [inline lib/new_note_screen/select_categories_screen.dart:7]: discuss too

File: lib/new_screen.dart [hunk 25]
```diff
@@ -0,0 +1,24 @@
+import 'package:flutter/material.dart';
+
+import 'bottom_navigation_bar.dart';
+
+class NewScreen extends StatefulWidget {
```
Comment (VovaMedvediev) [inline lib/new_screen.dart:5]: discuss

File: lib/provider/settings_notifier.dart [hunk 26]
```diff
@@ -0,0 +1,30 @@
+import 'dart:io';
+
+import 'package:flutter/material.dart';
+import 'package:flutter/services.dart';
+import 'package:image_picker/image_picker.dart';
+
+class SettingsNotifier with ChangeNotifier {
```
Comment (VovaMedvediev) [inline lib/provider/settings_notifier.dart:7]: change name. It seems to be ImageNotifier or NoteNotifier

File: lib/bottom_navigation_bar.dart (hunk 1 above)
Comment (katerynaYaryfa) [reply lib/bottom_navigation_bar.dart:6]: Done

File: lib/new_note_screen/add_new_note_screen.dart (hunk 2 above)
Comment (katerynaYaryfa) [reply lib/new_note_screen/add_new_note_screen.dart:11]: done

File: lib/new_note_screen/add_new_note_screen.dart (hunk 3 above)
Comment (katerynaYaryfa) [reply lib/new_note_screen/add_new_note_screen.dart:96]: done

File: lib/new_note_screen/add_new_note_screen.dart (hunk 4 above)
Comment (katerynaYaryfa) [reply lib/new_note_screen/add_new_note_screen.dart:104]: added

File: lib/new_note_screen/add_new_note_screen.dart (hunk 5 above)
Comment (katerynaYaryfa) [reply lib/new_note_screen/add_new_note_screen.dart:26]: deleted

File: lib/new_note_screen/photo_of_the_day_widget.dart (hunk 22 above)
Comment (katerynaYaryfa) [reply lib/new_note_screen/photo_of_the_day_widget.dart:208]: deleted

File: lib/provider/settings_notifier.dart (hunk 26 above)
Comment (katerynaYaryfa) [reply lib/provider/settings_notifier.dart:7]: done

File: lib/new_note_screen/how_was_your_day_widget.dart (hunk 11 above)
Comment (katerynaYaryfa) [reply lib/new_note_screen/how_was_your_day_widget.dart:13]: done

File: lib/new_note_screen/select_categories_screen.dart (hunk 24 above)
Comment (katerynaYaryfa) [reply lib/new_note_screen/select_categories_screen.dart:7]: discussed in dm

File: lib/new_screen.dart (hunk 25 above)
Comment (katerynaYaryfa) [reply lib/new_screen.dart:5]: discussed in dm

File: lib/new_note_screen/day_in_one_sentence_widget.dart (hunk 7 above)
Comment (katerynaYaryfa) [reply lib/new_note_screen/day_in_one_sentence_widget.dart:39]: done

File: lib/new_note_screen/day_in_one_sentence_widget.dart (hunk 8 above)
Comment (katerynaYaryfa) [reply lib/new_note_screen/day_in_one_sentence_widget.dart:44]: done

File: lib/new_note_screen/day_in_one_sentence_widget.dart (hunk 9 above)
Comment (katerynaYaryfa) [reply lib/new_note_screen/day_in_one_sentence_widget.dart:46]: done

File: lib/new_note_screen/day_in_one_sentence_widget.dart (hunk 10 above)
Comment (katerynaYaryfa) [reply lib/new_note_screen/day_in_one_sentence_widget.dart:71]: done

File: lib/new_note_screen/how_was_your_day_widget.dart (hunk 12 above)
Comment (katerynaYaryfa) [reply lib/new_note_screen/how_was_your_day_widget.dart:27]: done

File: lib/new_note_screen/photo_of_the_day_widget.dart (hunk 14 above)
Comment (katerynaYaryfa) [reply lib/new_note_screen/photo_of_the_day_widget.dart:24]: done

File: lib/new_note_screen/photo_of_the_day_widget.dart (hunk 15 above)
Comment (katerynaYaryfa) [reply lib/new_note_screen/photo_of_the_day_widget.dart:25]: done

File: lib/new_note_screen/photo_of_the_day_widget.dart (hunk 16 above)
Comment (katerynaYaryfa) [reply lib/new_note_screen/photo_of_the_day_widget.dart:26]: done

File: lib/new_note_screen/photo_of_the_day_widget.dart (hunk 17 above)
Comment (katerynaYaryfa) [reply lib/new_note_screen/photo_of_the_day_widget.dart:55]: done

File: lib/new_note_screen/photo_of_the_day_widget.dart (hunk 18 above)
Comment (katerynaYaryfa) [reply lib/new_note_screen/photo_of_the_day_widget.dart:80]: done

File: lib/new_note_screen/photo_of_the_day_widget.dart (hunk 19 above)
Comment (katerynaYaryfa) [reply lib/new_note_screen/photo_of_the_day_widget.dart:94]: done

File: lib/new_note_screen/photo_of_the_day_widget.dart (hunk 20 above)
Comment (katerynaYaryfa) [reply lib/new_note_screen/photo_of_the_day_widget.dart:189]: done

File: lib/new_note_screen/photo_of_the_day_widget.dart (hunk 21 above)
Comment (katerynaYaryfa) [reply lib/new_note_screen/photo_of_the_day_widget.dart:198]: done

File: lib/new_note_screen/how_was_your_day_widget.dart (hunk 13 above)
Comment (katerynaYaryfa) [reply lib/new_note_screen/how_was_your_day_widget.dart:57]: done

**Pull request merged**